portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -107,5 +123,43 @@ private Integer getMaxComments(HttpServletRequest request) {
     }
     return numMaxComments;
   }
+
+  //method from source code
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get(formInputElementName);
+
+    // User submitted form without selecting a file, so we can't get a URL. (dev server)
+    if (blobKeys == null || blobKeys.isEmpty()) {
+      return null;
+    }
+
+    // Our form only contains a single file input, so get the first index.
+    BlobKey blobKey = blobKeys.get(0);
+
+    // User submitted form without selecting a file, so we can't get a URL. (live server)
+    BlobInfo blobInfo = new BlobInfoFactory().loadBlobInfo(blobKey);
+    if (blobInfo.getSize() == 0) {
```

Comment:
Hmm does `getSize() == 0` mean they didn't upload a file at all, or that they uploaded a file with size 0? I'd be surprised if there's a `BlobInfo` at all if they didn't even select an upload (but have never used the API so I don't know!).

Reply:
I'm not entirely sure. I got the getUploadedFileURL() method from the source code provided in the walkthrough examples. I did a little digging and it seems this does check the size in bytes, so it seems this just checks the corner case if the file uploaded was somehow of size 0.

Here's the link I found this at: https://books.google.com/books?id=KX1ucolLqwMC&pg=PT445&lpg=PT445&dq=BlobInfo+Factory().load+BlobInfo(blob+Key)+size&source=bl&ots=sCCHwN-2w7&sig=ACfU3U1KL60tjlDzStpRnwoc6b9XN3cvzw&hl=en&sa=X&ved=2ahUKEwi9kJ2ovPrpAhWBdN8KHV1UC9UQ6AEwAXoECAoQAQ#v=onepage&q=getSize&f=false